airflow/plugins/operators/gtfs_csv_to_jsonl.py

```diff
@@ -0,0 +1,165 @@
+    @property
+    def successes(self) -> List[GTFSScheduleParseOutcome]:
+        return [outcome for outcome in self.outcomes if outcome.success]
+
+    @property
+    def failures(self) -> List[GTFSScheduleParseOutcome]:
+        return [outcome for outcome in self.outcomes if not outcome.success]
+
+    def save(self, fs):
+        self.save_content(
+            fs=fs,
+            content="\n".join(o.json() for o in self.outcomes).encode(),
+            exclude={"outcomes"},
+        )
+
+
+def parse_individual_file(
+    fs,
+    input_file: GTFSScheduleFeedFile,
+    gtfs_filename: str,
+) -> GTFSScheduleParseOutcome:
+    logging.info(f"Processing {input_file.path}")
+    field_names = []
+    lines = []
+    try:
+        with fs.open(input_file.path, newline="", mode="r") as f:
+            reader = csv.DictReader(f, restkey="calitp_unknown_fields")
+            field_names = reader.fieldnames
+            for row in reader:
+                lines.append(row)
+
+        jsonl_content = gzip.compress(
+            "\n".join(json.dumps(line) for line in lines).encode()
+        )
+
+        jsonl_file = GTFSScheduleFeedJSONL(
+            ts=input_file.ts,
+            base64_url=input_file.base64_url,
+            filename=gtfs_filename + ".jsonl.gz",
+            input_file_path=input_file.path,
+            gtfs_filename=gtfs_filename,
+        )
+
+        jsonl_file.save_content(content=jsonl_content, fs=fs)
+
+    except Exception as e:
+        logging.warn(f"Can't process {input_file.path}: {e}")
+        return GTFSScheduleParseOutcome(
+            success=False,
+            exception=e,
+            input_file_path=input_file.path,
+            fields=field_names,
+        )
+    logging.info(f"Parsed {input_file.path}")
+    return GTFSScheduleParseOutcome(
+        success=True,
+        input_file_path=input_file.path,
+        fields=field_names,
+        parsed_file=jsonl_file.path,
+    )
+
+
+def parse_files(day: pendulum.datetime, input_table: str, gtfs_filename: str):
+    fs = get_fs()
+    day = pendulum.instance(day).date()
+    files = fetch_all_in_partition(
+        cls=GTFSScheduleFeedFile,
+        bucket=SCHEDULE_UNZIPPED_BUCKET,
+        table=input_table,
+        fs=fs,
+        partitions={
+            "dt": day,
+        },
+        verbose=True,
+    )
+
+    logging.info(f"Identified {len(files)} records for {day}")
+    outcomes = []
+    for file in files:
+        outcome = parse_individual_file(fs, file, gtfs_filename)
+        outcomes.append(outcome)
+
+    result = ScheduleParseResult(filename="results.jsonl", dt=day, outcomes=outcomes)
+    result.save(fs)
+
+    assert len(files) == len(
+        result.outcomes
+    ), f"ended up with {len(outcomes)} outcomes from {len(files)} files"
+
+
+class GtfsGcsToJsonlOperator(BaseOperator):
+    def __init__(self, date, input_table, gtfs_filename, *args, **kwargs):
+        self.input_table = input_table
+        self.gtfs_filename = gtfs_filename
```

Comment:
is there a situation where these would ever differ except for removing the file extension?

Reply:
yeah, the case where something has a wrong name in the input but we want to map it onto the valid place. so like `_stops.txt` in input

Reply:
I don't know that this has actually happened, but it's something that this architecture specifically allows for

Reply:
Ok cool, I think providing a default could allow for less boilerplate but seems fine

Reply:
yeah fair, https://github.com/cal-itp/data-infra/pull/1738/commits/66eda24220ed159db29a4b9892b86aadc5b21984